### PR TITLE
Onboarding: Add tax task.

### LINKED ISSUE #2683: Onboarding - Task List: Set up tax
**User Story**
As a new merchant
I want some guidance in terms of setting up taxes
So I can set them up as quickly as possible

**Acceptance Criteria**
- If Jetpack and WC Services are installed, the store is connected and is located in a place that supports automatted taxes, the merchant can quickly enable taxes on this [screen](https://www.figma.com/file/xyKPRZ73kKhDyNjwbFvBJR/Task-List?node-id=976%3A8582)
- If the store location is unkown, the merchant needs to fill first the store location [here](https://www.figma.com/file/xyKPRZ73kKhDyNjwbFvBJR/Task-List?node-id=976%3A8623) before proceeding to setting up taxes
- If the store location doesn't support automatted taxes, the merchant sees a [screen](https://www.figma.com/file/xyKPRZ73kKhDyNjwbFvBJR/Task-List?node-id=976%3A8614) informing him that he needs to set them up manually. If the user clicks "Proceed" is redirected to the Tax Settings on WC Admin
- If the store is connected and located in a place that supports automatted taxes, but doesn't have Jetpack and WC Services installed, the merchant sees a [screen](https://www.figma.com/file/xyKPRZ73kKhDyNjwbFvBJR/Task-List?node-id=976%3A8601) to install the extensions for free.
- If the merchant wants to set up taxes manually instead of install the extensions, he's redirected to the [Tax Settings](https://www.figma.com/file/xyKPRZ73kKhDyNjwbFvBJR/Task-List?node-id=976%3A8658) on WC Admin
- If the merchant agrees to install the extensions, he then sees a [screen](https://www.figma.com/file/xyKPRZ73kKhDyNjwbFvBJR/Task-List?node-id=976%3A8660) asking him to connect the store to wp.com
- If the merchant clicks to connect the store, he's redirected to the [wp.com flow](https://www.figma.com/file/b0b2CrxPZz0i1aAPuv985tvK/Store-Profiler?node-id=1667%3A560) in the Profiler
- After the merchat connects the store to wp.com, he's redirected to this [screen ](https://www.figma.com/file/xyKPRZ73kKhDyNjwbFvBJR/Task-List?node-id=976%3A8566) with the automate sales tax calculation setting toggled on. If the user toggles the setting off and completes the task he needs to be redirected to the Tax Settings screen on WC Admin

**Links**
[Figma Mockup](https://www.figma.com/file/xyKPRZ73kKhDyNjwbFvBJR/Task-List?node-id=976%3A8530)

### PR DESCRIPTION
**Pull request by joshuatf**
Fixes #2683

Adds a task configuration task to the task list.

Note that this will require a follow-up to check for activated plugins and jetpack connection via the wc-api.

### Screenshots
Screen Shot 2019-08-22 at 5 37 53 PM
Screen Shot 2019-08-22 at 5 37 42 PM

### Detailed test instructions:

1. Delete your store address (or at least the `woocommerce_store_address` option), deactivate/uninstall wc-services and jetpack.
2. Go to the Tax task in the task list.
3.  Go through the steps and make sure you can fill in your address and install the 2 required plugins.
4.  The Jetpack wc-api connection will be added in a follow-up, for now toggle the value of `isJetpackConnected` in `tax.js` to bypass this step.
5.  Set `isJetpackConnected` to true and `pluginsToActivate` to an empty array.  Refresh this task and if you have all requirements met, you should see the success screen instead of the stepper.
6.  Make sure add a non-supported tax country forces you to manual configuration. https://developers.taxjar.com/api/reference/#countries
7. Make sure the toggle in the 4th step of the Stepper or the buttons in the success screen change the value of `wc_connect_taxes_enabled` in `wp_options`.  (Note: Jetpack and WCS must be installed and connected for this option to work via the WC REST API).

### CODE REVIEW

**Comment by rrennick** [comment]
@joshuatf It doesn't look like the tax step is included in the list of steps https://github.com/woocommerce/woocommerce-admin/blob/add/2683-task-tax/client/dashboard/profile-wizard/index.js#L30:L64 . Is it under one of the other steps?

**Comment by justinshreve** [comment]
@rrennick You need to open `dashboard/index.js` and set `requiredTasksComplete` to false. Otherwise, the task list will not display.

File: client/dashboard/task-list/tasks/tax.js [hunk 1]
```diff
@@ -0,0 +1,354 @@
...
+import { getAdminLink, getHistory, getNewPath } from '@woocommerce/navigation';
+
+/**
+ * Internal dependencies
+ */
+import Connect from './steps/connect';
+import Plugins from './steps/plugins';
+import StoreLocation from './steps/location';
+import withSelect from 'wc-api/with-select';
+import { Fragment } from '@wordpress/element/build/react';
```
[the remark this answers is not in the document]
Comment (joshuatf) [reply client/dashboard/task-list/tasks/tax.js]: Thanks for catching this- was automatically added by my editor.  Updated.

File: client/dashboard/task-list/tasks/tax.js [hunk 2]
```diff
@@ -0,0 +1,354 @@
...
+		const taxSettings = getSettings( 'tax' );
+		const isTaxSettingsError = Boolean( getSettingsError( 'tax' ) );
+		const isTaxSettingsRequesting = isGetSettingsRequesting( 'tax' );
+		// @todo This value should be fetched and updated via the wc-api.
+		const isJetpackConnected = false;
+		// @todo This should check against a list of already activated plugins and should be
+		// revisited after https://github.com/woocommerce/woocommerce-admin/pull/2825 is merged.
+		const pluginsToActivate = [ 'jetpack', 'woocommerce-services' ];
+
+		const countryCode = generalSettings.woocommerce_default_country
```
[the remark this answers is not in the document]
Comment (joshuatf) [reply client/dashboard/task-list/tasks/tax.js]: Good idea- added `getCountryCode()` to a dashboard utils file in https://github.com/woocommerce/woocommerce-admin/pull/2830/commits/9e3741d950e325c9832256c63ae21826b90528be

**Pull request merged**